### ISSUE TITLE
`model(X, labels=Y, return_dict=True).loss` is wrong

### ISSUE THREAD

**Issue by jettjaniak** [delphi-suite/delphi#133]
it should be `X, labels=X`
ideally we would force it to do what it was supposed to, instead of shifting tokens on it's own
but if we can't we need to adjust the design of tokenization script to produce sequences of seq_len (512) instead of seq_len+1 (513)

**Comment by jettjaniak** [comment]
we need some performance test to catch issues like this in the future

**Comment by jaidhyani** [comment]
Don't we still want to pass it seq_len+1? If it's converting it to internally we still get 512 positions on inputs of length 513, right?

https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L1233

**Comment by SrGonao** [comment]
I agree with jai, we don't need to change the tokenizer. @jaidhyani could you make a PR to fix this loss?

**Comment by jaidhyani** [comment]
Already merged it a few days ago. Still need to add performance tests though

**Comment by jettjaniak** [comment]
> Don't we still want to pass it seq_len+1? If it's converting it to internally we still get 512 positions on inputs of length 513, right?
> 
> https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L1233

I'm not sure what you're linking to, probably line numer shifted with some new commits on main. I believe it'll compute logits for all input ids, and then discard logits for last position when computing loss.

https://github.com/huggingface/transformers/blob/ccdabc5642bf84849af93f591e207dc625c8e1e1/src/transformers/models/llama/modeling_llama.py#L1213-L1214

It's not a big deal, but considering this I think inputs should be seq_len. LMK if you have strong takes
